ouroboros-consensus-byron/test/Test/ThreadNet/DualPBFT.hs

```diff
@@ -72,13 +73,40 @@ tests = testGroup "DualPBFT" [
 -- We limit it to 10 tests for now.
 prop_convergence :: SetupDualPBft -> Property
 prop_convergence setup = withMaxSuccess 10 $
+    (\prop -> if mightForgeInSlot0 then discard else prop) $
+    tabulate "Ref.PBFT result" [Ref.resultConstrName refResult] $
     prop_general
       (countByronGenTxs . dualBlockMain)
       (setupSecurityParam      setup)
-      (setupConfig             setup)
+      cfg
       (setupSchedule           setup)
+      (Just $ NumBlocks $ case refResult of
+         Ref.Forked{} -> 1
+         _            -> 0)
       (setupExpectedRejections setup)
+      1
       (setupTestOutput         setup)
+  where
+    cfg = setupConfig setup
+
+    refResult :: Ref.Result
+    refResult =
+      Ref.simulate (setupParams setup) (nodeJoinPlan cfg) (numSlots cfg)
+
+    -- The test infrastructure allows nodes to forge in slot 0; however, the
+    -- cardano-ledger-specs code causes @PBFTFailure (SlotNotAfterLastBlock
+    -- (Slot 0) (Slot 0))@ in that case. So we discard such tests.
```

Comment:
Could do with a comment that the spec here is therefore more strict than the real implementation.

Reply:
Elaborated in commit d0134cac7a0c684128b645903435d26412546c2f